### PR TITLE
compat for adr-97

### PR DESCRIPTION
**Pull request by nic547**
Compat for Adr-97
-Balistics
-Realistic Names

Possible other stuff?
- remove full auto from ps-90?
- add some diffrent ammo types?
- tweaking overheating?

### CODE REVIEW

**Comment by jonpas** [comment]
4 spaces for indentation please. :)

File: optionals/compat_adr_97/config.cpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#include "script_component.hpp"
+
+class CfgPatches {
+    class ADDON {
+        units[] = {};
+        weapons[] = {};
+        requiredVersion = REQUIRED_VERSION;
+        requiredAddons[] = {"A3_Weapons_F_Mod"};
+    author[]={"Nic547"};
```
Comment (jonpas) [inline optionals/compat_adr_97/config.cpp]: Indentation

File: optionals/compat_adr_97/CfgAmmo.hpp [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class CfgAmmo {
+    class BulletBase;
+    class B_570x28_Ball: BulletBase {
+        ACE_caliber=5.7;
```
Comment (jonpas) [inline optionals/compat_adr_97/CfgAmmo.hpp]: Spaces before and after `=` in the entire file.

**Comment by jonpas** [comment]
> remove full auto from ps-90?

I'd say yes.

> add some diffrent ammo types?

I don't think there is any other 5.7 ammo in Arma/ACE3 that is compatible with it. If custom ones are to be created then it can be in a separate PR. Though I'd say that's out of scope for an ACE3 compatibility anyways.

> tweaking overheating?

It currently uses Mk20 one, I think it's close enough, but if someone has numbers that can be done.

**Comment by nic547** [comment]
I'll do it as soon as i'm home (in a week)

**Comment by jonpas** [comment]
@nic547 status?

**Comment by nic547** [comment]
@jonpas should be finished now

File: optionals/compat_adr_97/CfgMagazines.hpp [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class CfgMagazines
+{
```
Comment (jonpas) [inline optionals/compat_adr_97/CfgMagazines.hpp]: Brace on same line.

File: optionals/compat_adr_97/CfgMagazines.hpp [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+	class CA_Magazine;
+	class 50Rnd_570x28_SMG_03: CA_Magazine
+	{
```
Comment (jonpas) [inline optionals/compat_adr_97/CfgMagazines.hpp]: Brace on same line.

File: optionals/compat_adr_97/CfgMagazines.hpp [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+		displayName = CSTRING(P90_Mag_Name);
+        descriptionShort = CSTRING(P90_Mag_DescriptionShort);
```
Comment (jonpas) [inline optionals/compat_adr_97/CfgMagazines.hpp]: Indentation quirk.

File: optionals/compat_adr_97/CfgWeapons.hpp [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+    class Rifle_Base_F;
+    class SMG_03_TR_BASE: Rifle_Base_F {
+        ACE_barrelTwist = 228.6; // 1:9 inch twist
+        ACE_barrelLenght = 407;
+        ACE_twistDirection = 1;
+        modes[]=
+		{
+			"Single"
+		};
```
Comment (jonpas) [inline optionals/compat_adr_97/CfgWeapons.hpp]: This section should be indented correctly and brace on same line.

File: optionals/compat_adr_97/CfgWeapons.hpp [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    class SMG_03_hex: SMG_03_black {
+        displayName = CSTRING(PS90_Hex_Name);
+    };
+    class SMG_03C_BASE: SMG_03_TR_BASE {
+        ACE_barrelLenght = 264;
+        modes[]=
+		{
+			"Single",
+			"FullAuto"
+		};
```
Comment (jonpas) [inline optionals/compat_adr_97/CfgWeapons.hpp]: Same as above.

File: optionals/compat_adr_97/stringtable.xml [hunk 8]
```diff
@@ -0,0 +1,221 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project name="ACE">
+    <Package name="ace_compat_adr_90">
```
Comment (jonpas) [inline optionals/compat_adr_97/stringtable.xml]: Should be `Compat_ADR_97`.

And every Key ID should be `STR_ACE_Compat_ADR_97_*`.

File: optionals/compat_adr_97/script_component.hpp [hunk 9]
```diff
@@ -0,0 +1,5 @@
+#define COMPONENT Compat_ADR_97
```
Comment (jonpas) [inline optionals/compat_adr_97/script_component.hpp]: This was okay as it was. :)

**Comment by jonpas** [comment]
:+1:

**Pull request merged**